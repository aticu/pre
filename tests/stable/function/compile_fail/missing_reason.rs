use pre::pre;

#[pre("must be bar")]
fn foo() {}

#[pre]
fn main() {
    #[assure("must be bar")]
    foo()
}
