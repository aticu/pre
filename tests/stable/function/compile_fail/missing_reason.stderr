error: you need to specify a reason why this precondition holds

  = help: add `, reason = "<specify the reason why you can assure this here>"`

 --> $DIR/missing_reason.rs:8:14
  |
8 |     #[assure("must be bar")]
  |              ^^^^^^^^^^^^^
